odd-platform-api/src/main/java/com/provectus/oddplatform/service/DataEntityServiceImpl.java

```diff
@@ -79,10 +97,16 @@ public DataEntityServiceImpl(final DataEntityMapper entityMapper,
     @Override
     public Mono<DataEntityDetails> getDetails(final long dataEntityId) {
         return Mono.fromCallable(() -> entityRepository.getDetails(dataEntityId))
-            .flatMap(optional -> optional.isEmpty()
-                ? Mono.error(new NotFoundException())
-                : Mono.just(optional.get()))
-            .map(entityMapper::mapDtoDetails);
+                .flatMap(optional -> optional.isEmpty()
+                        ? Mono.error(new NotFoundException())
+                        : Mono.just(optional.get()))
+                .map(entityMapper::mapDtoDetails)
+                .map(this::incrementViewCount);
+    }
+
+    public DataEntityDetails incrementViewCount(DataEntityDetails dto) {
+        entityRepository.incrementViewCount(dto);
```

Comment:
idk if there's a point in this method coz it just makes a call and returns the same dto back again, but I didn't want to make repo's method to return the same dto without any reason, so it's void there.

Reply:
I believe it's fine as it is.
Please change signature to private

Reply:
However this approach will return outdated VIEW_COUNT field, since you update the count after you map it to OpenAPI entity

Reply:
Fixed. Now incrementing the count in a dto after a repo method call.